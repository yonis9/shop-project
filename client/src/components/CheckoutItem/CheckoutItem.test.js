import React from 'react';
import { shallow } from 'enzyme';

import { CheckoutItem } from './CheckoutItem';

describe('CheckoutItem component', () => {
    let wrapper;
    let mockClearItem;
    let mockAddItem;
    let mockRemoveItem;

    beforeEach(() => {
        mockAddItem = jest.fn();
        mockRemoveItem = jest.fn();
        mockClearItem = jest.fn();

        const mockCartItem = {
            name: 'shirt',
            imageUrl: 'shirts.com',
            price: 15,
            quantity: 2
        };

        const mockProps = {
            cartItem: mockCartItem,
            addItem: mockAddItem,
            removeItem: mockRemoveItem,
            clearItem: mockClearItem
        };

        wrapper = shallow(<CheckoutItem {...mockProps} />);
    });

    it('should rendered CheckoutItem component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should call clearItem when remove button is clicked', () => {
        wrapper.find('RemoveButtonContainer').simulate('click');
        expect(mockClearItem).toHaveBeenCalled();
    })

    it('should call removeItem when left arrow is clicked', () => {
        wrapper.find('QuantityContainer').childAt(0).simulate('click');
        expect(mockRemoveItem).toHaveBeenCalled();
    })

    it('should call addItem when right arrow is clicked', () => {
        wrapper.find('QuantityContainer').childAt(2).simulate('click');
        expect(mockAddItem).toHaveBeenCalled();
    })

});