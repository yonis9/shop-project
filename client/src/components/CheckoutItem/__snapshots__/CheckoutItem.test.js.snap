// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component should rendered CheckoutItem component 1`] = `ShallowWrapper {}`;
